library/src/main/java/com/google/android/exoplayer2/text/eia608/Eia608Decoder.java

```diff
@@ -414,32 +501,197 @@ private void handleMiscCode(byte cc2) {
 
     switch (cc2) {
       case CTRL_ERASE_DISPLAYED_MEMORY:
-        captionString = null;
         if (captionMode == CC_MODE_ROLL_UP || captionMode == CC_MODE_PAINT_ON) {
-          captionStringBuilder.setLength(0);
+          flushCaptionBuilder();
         }
         return;
       case CTRL_ERASE_NON_DISPLAYED_MEMORY:
-        captionStringBuilder.setLength(0);
+        flushCaptionBuilder();
         return;
       case CTRL_END_OF_CAPTION:
-        captionString = getDisplayCaption();
-        captionStringBuilder.setLength(0);
+        buildCue();
+        flushCaptionBuilder();
         return;
       case CTRL_CARRIAGE_RETURN:
         maybeAppendNewline();
         return;
       case CTRL_BACKSPACE:
-        if (captionStringBuilder.length() > 0) {
-          captionStringBuilder.setLength(captionStringBuilder.length() - 1);
-        }
+        backspace();
         return;
     }
   }
 
+  private void handlePreambleCode(byte cc1, byte cc2) {
+    // For PAC layout see: https://en.wikipedia.org/wiki/EIA-608#Control_commands
+    applySpan(); // Apply any spans.
+
+    // Parse the "next row down" flag.
+    boolean nextRowDown = (cc2 & 0x20) != 0;
+    if (nextRowDown) {
+      // TODO: We should create a new cue instead, this may cause issues when
+      // the new line receives it's own PAC which we ignore currently.
+      // As a result of that the new line will be positioned directly below the
+      // previous line.
+      maybeAppendNewline();
```

Comment:
Ideally, we should create a new Cue here, but I'm not sure how we can display Cues within different Eia608Subtitle objects on the screen at the same time. Maybe we shouldn't try to instantiate new Eia608Subtitle objects in `dequeueOutputBuffer()` all the time, and instead stream the cues?